### PR TITLE
Metabolic Syndrome Prediction | 5. Model prediction and evaluation

### LINKED ISSUE #98: Metabolic Syndrome Prediction | 5. Model prediction and evaluation
### Is your feature request related to a problem? Please describe.

After all data play, let's delve into actual training part.

### Describe the solution you'd like

1. Write generic functions.
2. Use all features; no feature engineering (transformation, selection, extraction, dropping here)
3. Use both normalized and standardized data.
4. Use hyperparameter tuning for models.
5. Use some training strategy like LOSO or K-Fold.
6. Record time to find best params and time to train using those params.
7. Predict on test data and print all eval metrics.
8. If it is binary classification, plot ROC to find optimal threshold.
9. Using that, plot Confusion matrix to evaluate the models.
10. Models to use: Extra Trees, Extra Tree, SVM, Logistic, KNN, DT, RF, GB, Bagging Clf, XGB, Hist GB, MLP, Catboost, Adaboost, Naive Bayes, LightGBM

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Arihant-Bhandari**
closes issue #98 Metabolic Syndrome Prediction | 5. Model prediction and evaluation

worked on step 5:

1. Write generic functions.

2. Use all features; no feature engineering (transformation, selection, extraction, dropping here)

3. Use both normalized and standardized data.

4. Use hyperparameter tuning for models.

5. Use some training strategy like LOSO or K-Fold.

6. Record time to find best params and time to train using those params.

7.  on test data and print all eval metrics.

8. If it is binary classification, plot ROC to find optimal threshold.

9. Using that, plot Confusion matrix to evaluate the models.

10. Models to use: Extra Trees, SVM, Logistic, KNN, DT, RF, GB, Bagging Clf, XGB, Hist GB, MLP, Catboost, Adaboost, Naive Bayes, LightGBM

Worked on the following models:

1) XGBoost Classifier
2) Random Forest Classifier
3) Decision Tree Classifier
4) AdaBoost Classifier
5) LGBM (Light Gradient Boosting Machine) Classifier
6) Logistic Regression
7) Extra Trees Classifier
8) Histogram-based Gradient Boosting Classifier
9) Gradient Boosting Classifier
10) KNN (K-Nearest Neighbors)
11) Stacking Classifier
12) Voting Classifier
13) Bagging Classifier
14) CatBoost Classifier
15) Linear SVC (Support Vector Classifier)
16) SVC (Support Vector Classifier)
17) Perceptron
18) Multilayer Perceptron (Feedforward ANN)
19) Artificial Neural Networks

### CODE REVIEW

**Comment by Arihant-Bhandari** [comment]
hi @SrijanShovit this is the initial work i hope to get feedback on, from the problem statement i have devised stuff for XGBClassifier, if this work is found satisfactory i will continue with this for all other models, pls check up on this. if anything is to be added or changed, i will be glad to do them.

thank you for your time and patience.

**Review by SrijanShovit — CHANGES_REQUESTED** [review]
Process of a model prediction is fine but repeat same thing for various models.

In the end, use 2-3 neural networks as well.

**Comment by Arihant-Bhandari** [comment]
Hi @SrijanShovit , the work on this PR is complete, pls review and if any suggestions or changes are to be made, please let me know, will gladly make the required changes.
